### PR TITLE
Make actions a bit quicker

### PR DESCRIPTION
**Pull request by retsyx**
Highlight first tip in tooltip to allow immediately pressing enter to quickly select it.
If there is only one result, don't bother displaying the tooltip, just launch the action - especially useful for jira actions if the default google action is removed.

### CODE REVIEW

**Comment by tanin47** [comment]
Thank you @retsyx. I think the improvement makes sense.

I'd like this feature to be more configurable, as Tip is a personal productivity tool (so it should be configurable in a way that an individual wants). 

Therefore, I'd like to ask for more changes:

* We can add a new optional field to the JSON: `executeIfOnlyOne` (default to `false`). If this field is `true`, then we will auto-execute the action if it's the only action. This will let end-users choose if they want to do this or not
* Please also add a test case for this case as well.

Thank you!

**Comment by tanin47** [comment]
@retsyx please assign it back to me when you finish. Thank you!

**Comment by retsyx** [comment]
It doesn't look like I'm able to assign it. It's done.

File: mac-app/TipUITests/TipUITests.swift [hunk 1]
```diff
@@ -35,6 +35,20 @@ final class TipUITests : XCTestCase {
     XCTAssertEqual("tanintip://TestInput", pasteBoard.string(forType: .string))
   }
 
+  func testSingleProviderClickingOnTextWithLabel() {
```
Comment (tanin47) [inline mac-app/TipUITests/TipUITests.swift:38]: Thank you for this extra test.

File: mac-app/Tip/TipTableController.m [hunk 2]
```diff
@@ -68,6 +68,9 @@ - (void)setItems:(NSArray<TipItem *> *)items {
     _error = nil;
     [self update];
     [_table reloadData];
+    if (items.count > 0) {
```
Comment (tanin47) [inline mac-app/Tip/TipTableController.m:71]: This is nice. Thank you!

**Review by tanin47 — APPROVED** [review]
Thank you for your contribution!

**Comment by tanin47** [comment]
@retsyx we have an RC for 1.2.1, which includes this improvement. I'd love your feedback before I release the stable version https://github.com/tanin47/tip/releases/tag/v1.2.1-rc1

I changed the implementation a little. I was concerned that there was no visual feedback, so I added one.

**Comment by retsyx** [comment]
Thanks for asking.

Personally, for me, I find that the action itself (my browser jumping and opening a page) is enough feedback. The extra delay for the visual feedback hinders my particular use case. I imagine it is more important to have the visual feedback for clipboard actions.

My $.2

BTW, I noticed that when I clicked one of two available clipboard actions (e.g. invoke shortcut on '2342' in Terminal with your default date actions in the script, click one of the dates), the popup did not dismiss.

**Comment by tanin47** [comment]
> I imagine it is more important to have the visual feedback for clipboard actions.

You're right. It's actually only important for copying to clipboard use case. Opening a URL already has its own sufficient feedback. 

That's a good callout. I'll fix this.

> BTW, I noticed that when I clicked one of two available clipboard actions (e.g. invoke shortcut on '2342' in Terminal with your default date actions in the script, click one of the dates), the popup did not dismiss.

Yes, this is expected. Thinking about it, dismissing it would have been better. Users can always re-trigger the tooltip anyway.

**Comment by tanin47** [comment]
@retsyx v1.3.0-rc1 has been released with this fix. Could you try to see if this is what you expect? https://github.com/tanin47/tip/releases/tag/v1.3.0-rc1 Thank you!

**Comment by retsyx** [comment]
Looks good!

**Pull request merged**